src/test/scala/htsjdk/samtools/util/StringUtilTest.scala

```diff
@@ -0,0 +1,111 @@
+package htsjdk.samtools.util
+
+import htsjdk.UnitSpec
+
+class StringUtilTest extends UnitSpec {
+  "StringUtil.split" should "should behave like String.split(char)" in {
+    Seq("A:BB:C", "A:BB", "A:BB:", "A:BB:C:DDD", "A:", "A", "A:BB:C").foreach { s =>
+      val arr = new Array[String](10)
+      val count = StringUtil.split(s, arr, ':')
+      arr.take(count) shouldBe s.split(':')
+    }
+  }
+
+  "StringUtil.splitConcatenateExcessTokens" should "behave like String.split(regex, limit)" in {
+    Seq("A:BB:C", "A:BB", "A:BB:", "A:BB:C:DDD", "A:", "A", "A:BB:C:").foreach { s =>
+      val arr = new Array[String](3)
+      val count = StringUtil.splitConcatenateExcessTokens(s, arr, ':')
+      arr.take(count) shouldBe s.split(":", 3).filter(_.nonEmpty)
+    }
+  }
+
+  "StringUtil.join" should "join tokens with a separator" in {
+    StringUtil.join(",", 1, "hello", 'T') shouldBe "1,hello,T"
+    StringUtil.join(",") shouldBe ""
+  }
+
+  "StringUtil.hammingDistance" should "return zero for two empty sequences" in {
+      StringUtil.hammingDistance("", "") shouldBe 0
+  }
+
+  Seq(("ATAC", "GCAT", 3), ("ATAGC", "ATAGC", 0)).foreach { case (s1, s2, distance) =>
+      it should s"return distance ${0} between $s1 and $s2" in {
```

Comment:
also, why did you switch the order between the data and the test? everywhere else it's "test" in "data and here it's data => test.... I don't mind...I just want to understand if it matters...